.github/workflows/test-canary.yml

```diff
@@ -0,0 +1,105 @@
+# This pipeline purpose is solely meant to run a subset of our test suites against upcoming or unreleased dependencies versions
+name: test-canary
+
+on:
+  push:
+    branches:
+      - main
+      - 'release/**'
+  pull_request:
+    paths-ignore:
+      - '**.md'
+
+env:
+  # If left blank, will default to latest stable
+  GO_VERSION: "1.23-rc1"
+  GO_VERSION_SETUP_GO: "1.23.0-rc.1"
```

Comment:
Maybe parse https://go.dev/dl/?mode=json&include=all to automatically get the latest version

Reply:
I like that. Will do.